### PR TITLE
[Testing:System] Cypress test for Profile Page

### PR DESCRIPTION
**Pull request by Yujjio**
### Please check if the PR fulfills these requirements:

* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant

### What is the current behavior?
Currently no cypress e2e test for the user's profile page

### What is the new behavior?
created basic tests, including:
1. check essential info columns' visibility
2. check pop-up forms' visibility
3. randomly generate updated info and check whether the database was updated

### CODE REVIEW

File: site/cypress/e2e/profile.spec.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        result += characters.charAt(Math.floor(Math.random() * characters.length));
+        counter += 1;
+    }
+    return result;
+};
+
+const given = makeid();
+const family = makeid();
+const pronouns = makeid();
+var mail = makeid();
```
Comment (hansongu123) [inline site/cypress/e2e/profile.spec.js:26]:
```suggestion
let mail = makeid();
```

File: site/cypress/e2e/profile.spec.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    it('Verify every pop-up form', () => {
+        cy.visit('/user_profile');
+        cy.login();
+
+        // verify that every form can be intrigued
+        cy.get('.fa-pencil-alt').should(($icon) => {
+            expect($icon).to.have.length(5);
+        });
+
+        // verify prederred name form
```
Comment (hansongu123) [inline site/cypress/e2e/profile.spec.js:57]:
```suggestion
        // verify preferred name form
```

**Review by hansongu123 — COMMENTED** [review]
Tests overall look reasonable and have a great coverage. Some minor changes are requested.  I'm a little bit unsure about the for loop logic here.

**Review by williamjallen — APPROVED** [review]
Changes seem reasonable to me

**Pull request merged**